client/packages/host/src/utils.ts

```diff
@@ -1,7 +1,24 @@
-import { ReactNode } from 'react';
+import { ReactNode, useState, useEffect } from 'react';
 
 export const getBadgeProps = (value?: number) => ({
   badgeContent: (value ?? 0) as ReactNode,
   max: 99,
   color: 'primary' as 'primary' | 'default',
 });
+
+export const useDeviceDetect = () => {
+  const [isMobile, setMobile] = useState(false);
+
+  useEffect(() => {
+    const userAgent =
+      typeof window.navigator === 'undefined' ? '' : navigator.userAgent;
+    const mobile = Boolean(
+      userAgent.match(
+        /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
+      )
+    );
+    setMobile(mobile);
+  }, []);
+
+  return { isMobile };
+};
```

Comment:
Since `window.navigator.userAgent` is just a static value that is always available, I'm not sure this needs to be a hook (i.e. no need to hold any of its own state). Just a simple expression will do, I think:

```ts
export const isMobile = Boolean(
  navigator.userAgent.match(
    /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
  )
);
```

Also, a better location for this might be `client/packages/config/src/config.ts`, though perhaps @mark-prins might have another suggestion?